lms/djangoapps/course_blocks/transformers/split_test.py

```diff
@@ -0,0 +1,110 @@
+"""
+Split Test Block Transformer, used to filter course structure per user.
+"""
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+from .helpers import get_user_partition_groups
+
+
+class SplitTestTransformer(BlockStructureTransformer):
+    """
+    Split Test Transformer Class
+    """
+    VERSION = 1
+
+    @staticmethod
+    def check_split_access(split_test_groups, user_groups):
+        """
+        Check that user has access to specific split test group.
+
+        Arguments:
+            split_test_groups (list)
+            user_groups (dict[Partition Id: Group])
+
+        Returns:
+            bool
+        """
+        if split_test_groups:
+            for _, group in user_groups.iteritems():
+                if group.id in split_test_groups:
+                    return True
+            return False
+        return True
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Computes any information for each XBlock that's necessary to execute
+        this transformer's transform method.
+
+        Arguments:
+            block_structure (BlockStructureCollectedData)
+        """
+
+        # Check potential previously set values for user_partitions and split_test_partitions
+        xblock = block_structure.get_xblock(block_structure.root_block_key)
+        user_partitions = getattr(xblock, 'user_partitions', [])
+        split_test_partitions = getattr(xblock, 'split_test_partition', []) or []
```

Comment:
@dino-cikatic Is the `split_test_partition` field set on an `xblock`?  I don't see that field in https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/split_test_module.py